From 0891afb74f53b1b6f836df94741aa91110ee0808 Mon Sep 17 00:00:00 2001
From: ChenXiaoSong <chenxiaosongemail@foxmail.com>
Date: Sun, 1 May 2022 23:21:14 +0800
Subject: [PATCH] x86_64: gcc build with -Og.

make O=build olddefconfig -j64 && make O=build bzImage -j64 && make O=build modules -j64 && make O=build modules_install INSTALL_MOD_PATH=mod -j64

If dump_stack() is all ?, then revert "x86/unwind/orc: Don't skip the first frame for inactive tasks".

Signed-off-by: ChenXiaoSong <chenxiaosongemail@foxmail.com>
---
 Makefile                            | 9 ++-------
 include/linux/compiler_attributes.h | 2 +-
 include/linux/compiler_types.h      | 4 ++--
 3 files changed, 5 insertions(+), 10 deletions(-)

diff --git a/Makefile b/Makefile
index cfbe6a7de640..9d9b0263b076 100644
--- a/Makefile
+++ b/Makefile
@@ -820,13 +820,8 @@ KBUILD_CFLAGS	+= $(call cc-disable-warning, format-truncation)
 KBUILD_CFLAGS	+= $(call cc-disable-warning, format-overflow)
 KBUILD_CFLAGS	+= $(call cc-disable-warning, address-of-packed-member)
 
-ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE
-KBUILD_CFLAGS += -O2
-KBUILD_RUSTFLAGS += -Copt-level=2
-else ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
-KBUILD_CFLAGS += -Os
-KBUILD_RUSTFLAGS += -Copt-level=s
-endif
+KBUILD_CFLAGS += -Og
+KBUILD_RUSTFLAGS += -Copt-level=g
 
 # Always set `debug-assertions` and `overflow-checks` because their default
 # depends on `opt-level` and `debug-assertions`, respectively.
diff --git a/include/linux/compiler_attributes.h b/include/linux/compiler_attributes.h
index 898b3458b24a..c60f46997481 100644
--- a/include/linux/compiler_attributes.h
+++ b/include/linux/compiler_attributes.h
@@ -144,7 +144,7 @@
  *   gcc: https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#index-error-function-attribute
  */
 #if __has_attribute(__error__)
-# define __compiletime_error(msg)       __attribute__((__error__(msg)))
+# define __compiletime_error(msg) //       __attribute__((__error__(msg)))
 #else
 # define __compiletime_error(msg)
 #endif
diff --git a/include/linux/compiler_types.h b/include/linux/compiler_types.h
index eb0466236661..28396bd19ac4 100644
--- a/include/linux/compiler_types.h
+++ b/include/linux/compiler_types.h
@@ -353,8 +353,8 @@ struct ftrace_likely_data {
  * supplied condition is *false*, emitting the supplied error message if the
  * compiler has support to do so.
  */
-#define compiletime_assert(condition, msg) \
-	_compiletime_assert(condition, msg, __compiletime_assert_, __COUNTER__)
+#define compiletime_assert(condition, msg)
+// 	_compiletime_assert(condition, msg, __compiletime_assert_, __COUNTER__)
 
 #define compiletime_assert_atomic_type(t)				\
 	compiletime_assert(__native_word(t),				\
-- 
2.34.1

